agcaballero/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -35,144 +46,130 @@
     Collection<String> attendees = request.getAttendees();
     Collection<String> optionalAttendees = request.getOptionalAttendees();
     
-    // are there any mandatory employees?
-    boolean mandatory = false;
-    if(attendees.size() > 0) {
-      mandatory = true;
-    }
-    
-    // are there any optional employees?
-    boolean optional = false; 
-    if(optionalAttendees.size() > 0) {
-      optional = true;
-    }
+    // Holds whether the mandatory attendees can attend & the number of optional attendees 
+    // that can attend. If minuteStatus[number] is equal to MANDATORY_UNAVAILABLE, then mandatory
+    // attendees can't attend at that number. If it's equal to 0, then all attendees can attend.
+    // If it's equal to a negative int, then the abs value of that negative number is the number 
+    // of optional attendees who can't attend. The +1 is to account for last minute in day
+    int[] minuteStatus = new int[MINUTES_IN_DAY + 1];
 
-    // minutes of day is + 1 to account for first and last minute of day
-    int[] minutes = new int[MINUTES_IN_DAY + 1];
-    // 0 - all attendees (mandatory & optional) can attend 
-    // 1 - mandatory attendees can attend (but not all optional)
-    // 2 - not all mandatory attendees can attend (even if all optional can)
+    // keep track of min status reached 
+    int minStatus = 0;
```

Comment:
Reading this variable name and comment above, I'm still not sure what this variable is for. Is there a potentially more descriptive name? Longer is fine, clarity is more important than saving horizontal space.

Reply:
It keeps track of the lowest status found so that when you look through the array with a status in mind, it'll know when to stop. For example, if the lowest value in the minuteStatus array is -7, then that champion will be -7, and we'll know that if we don't find availabilities where all seven people can't attend (after checking all the previous higher statuses like -6, -5, -4) then we'll have to return no availability. I changed the name to ```lowestStatusFound``` and updated the comment to hopefully make this more clear.

Reply:
Done